paddle/go/recordio/multi_reader.go

```diff
@@ -0,0 +1,140 @@
+package recordio
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+// MultiScanner is a scanner for multiple recordio files.
+type MultiScanner struct {
```

Comment:
We'd rename this high-level API to `Scanner`. And we rename the existing `Scanner` to `RangeScanner`. We can do this in a future PR.